src/app/process-editor/components/process-modeler/palette/palette.component.ts

```diff
@@ -0,0 +1,87 @@
+ /*!
+ * @license
+ * Copyright 2019 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Component, Inject, Optional, HostListener, ViewChild, TemplateRef } from '@angular/core';
+import { ProcessModelerPaletteService } from '../../../services/palette/process-modeler-palette.service';
+import { PaletteElement, PaletteElementsToken, ToolTrigger } from 'ama-sdk';
+import { OverlayRef } from '@angular/cdk/overlay';
+import { PaletteOverlayDirective } from './palette-overlay.directive';
+
+@Component({
+    templateUrl: './palette.component.html',
+    selector: 'ama-process-palette'
+})
+export class PaletteComponent {
+
+    public selectedTool: ToolTrigger;
+    public paletteElements: PaletteElement[];
+    public opened = true;
+    public detach = false;
+    overlayRef: OverlayRef;
+    @ViewChild('drawer') templateContent: TemplateRef<any>;
```

Comment:
Hmm, can it handle multiple submenus also? :O
Because that one is a valid scenario also.

Reply:
It should work, I tested this scenario.